ecosystem/typescript/sdk_v2/src/bcs/move-types/structs.ts

```diff
@@ -0,0 +1,401 @@
+   * @params values: an array of `numbers` to convert to Bools
+   * @returns a `Vector<Bool>`
+   */
+  static Bool(values: Array<boolean>): Vector<Bool> {
+    return new Vector<Bool>(values.map((v) => new Bool(v)));
+  }
+
+  /**
+   * Factory method to generate a Vector of MoveStrings from an array of strings.
+   *
+   * @example
+   * const v = Vector.String(["hello", "world"]);
+   * @params values: an array of `numbers` to convert to MoveStrings
+   * @returns a `Vector<MoveString>`
+   */
+  static String(values: Array<string>): Vector<MoveString> {
+    return new Vector<MoveString>(values.map((v) => new MoveString(v)));
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeVector(this.values);
+  }
+
+  /**
+   * Deserializes an `deserializer: Deserializer` instance to a Vector of type T, where T is a
+   * class that implements Serializable and Deserializable.
+   *
+   * If you want to deserialize types that are merely Deserializable into an Array<T> instead
+   * of a Vector<T>, use the `deserializeVector` function in the Deserializer class.
+   *
+   * To be clear: this function `Vector.deserialize<T>(...)` returns a `Vector<T>`, whereas
+   * `deserializeVector<U>(...)` returns an `Array<U>`.
+   *
+   * NOTE: Currently, this function only returns correct values with a generic type depth of one.
+   * Complex generics will not deserialize as expected, because Typescript does not have generic
+   * type information at runtime.
+   *
+   * For example, `Vector.deserialize(U8)` will correctly deserialize a byte vector to a `Vector<U8>`,
+   * but if you try `Vector.deserialize(Vector<U8>)` to get a `Vector<Vector<u8>>`, it will not work.
+   *
+   * @example
+   * const vec = Vector.deserialize(deserializer, U64);
+   * @params deserializer: the Deserializer instance to use, with bytes loaded into it already.
+   * cls: the class to typecast the input values to, must be a Serializable and Deserializable type.
+   * @returns a Vector of the corresponding class T
+   * *
+   */
+  static deserialize<T extends Serializable>(deserializer: Deserializer, cls: Deserializable<T>): Vector<T> {
+    const length = deserializer.deserializeUleb128AsU32();
+    const values = new Array<T>();
+    for (let i = 0; i < length; i += 1) {
+      values.push(cls.deserialize(deserializer));
+    }
+    return new Vector(values);
+  }
+}
+
+export class MoveString extends Serializable {
+  public value: string;
+
+  constructor(value: string) {
+    super();
+    this.value = value;
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeStr(this.value);
+  }
+
+  static deserialize(deserializer: Deserializer): MoveString {
+    return new MoveString(deserializer.deserializeStr());
+  }
+}
+
+export class MoveOption<T extends Serializable> extends Serializable {
+  private vec: Vector<T>;
+
+  public value: T | undefined;
+
+  constructor(value?: T) {
+    super();
+    if (typeof value !== "undefined") {
```

Comment:
if making the suggested change (see above) need to change it here to also check for `null`